pandas/tests/libs/test_hashtable.py

```diff
@@ -440,6 +441,27 @@ def test_nan_in_nested_tuple(self):
             table.get_item(other)
         assert str(error.value) == str(other)
 
+    def test_nan_in_namedtuple(self):
+        T = namedtuple("T", ["x"])
+        nan1 = T(float("nan"))
+        nan2 = T(float("nan"))
+        assert nan1.x is not nan2.x
+        table = ht.PyObjectHashTable()
+        table.set_item(nan1, 42)
+        assert table.get_item(nan2) == 42
+
+    def test_nan_in_nested_namedtuple(self):
+        T = namedtuple("T", ["x", "y"])
+        nan1 = T(1, (2, (float("nan"),)))
+        nan2 = T(1, (2, (float("nan"),)))
+        other = T(1, 2)
+        table = ht.PyObjectHashTable()
+        table.set_item(nan1, 42)
+        assert table.get_item(nan2) == 42
+        with pytest.raises(KeyError, match=None) as error:
+            table.get_item(other)
+        assert str(error.value) == str(other)
```

Comment:
Instead could you use the `match` argument instead of setting it to `None`?

Reply:
This line should be removed now that you added the match argument

Reply:
Yeah, that makes more sense. I'll fix the other tests too while I'm at it.